src/App/FreeCADInit.py

```diff
@@ -192,6 +192,32 @@ def RunInitPy(Dir):
         else:
             Log('Init:      Initializing ' + Dir + '(Init.py not found)... ignore\n')
 
+    def processMetadataFile(MetadataFile):
+        meta = FreeCAD.Metadata(MetadataFile)
+        if not meta.supportsCurrentFreeCAD():
+            Msg(f'NOTICE: {meta.Name} does not support this version of FreeCAD, so is being skipped\n')
+            return None
+        content = meta.Content
+        if "workbench" in content:
+            workbenches = content["workbench"]
+            for workbench in workbenches:
+                if not workbench.supportsCurrentFreeCAD():
+                    Msg(f'NOTICE: {meta.Name} content item {workbench.Name} does not support this version of FreeCAD, so is being skipped\n')
+                    return None
+                subdirectory = workbench.Name if not workbench.Subdirectory else workbench.Subdirectory
+                subdirectory = subdirectory.replace("/",os.path.sep)
+                subdirectory = os.path.join(Dir, subdirectory)
+                #classname = workbench.Classname
+                sys.path.insert(0,subdirectory)
+                PathExtension.append(subdirectory)
+                RunInitPy(subdirectory)
+
+    def tryProcessMetadataFile(MetadataFile):
+        try:
+            processMetadataFile(MetadataFile)
+        except Exception as exc:

```

Comment:
I'm surprised the linter isn't yelling about this -- is there a narrower exception we can catch?

Reply:
`Exception` is used in other places of these two files, too. But the real exception type is the custom type `Base.XMLBaseException`